File: advertising_sdk/src/main/java/com/adadapted/android/sdk/core/atl/AddToListContent.java
package com.adadapted.android.sdk.core.atl;

import java.util.List;

public interface AddToListContent {
    final class Sources {
        public static final String IN_APP = "in_app";
        public static final String OUT_OF_APP = "out_of_app";
    }

    void acknowledge();
    void itemAcknowledge(AddToListItem item);
    void failed(String message);
    void itemFailed(AddToListItem item, String message);
    String getSource();
    List<AddToListItem> getItems();
    boolean hasNoItems();
}

File: advertising_sdk/src/main/java/com/adadapted/android/sdk/core/atl/AddToListContent.kt
package com.adadapted.android.sdk.core.atl

interface AddToListContent {
    object Sources {
        const val IN_APP = "in_app"
        const val OUT_OF_APP = "out_of_app"
    }

    fun acknowledge()
    fun itemAcknowledge(item: AddToListItem)
    fun failed(message: String)
    fun itemFailed(item: AddToListItem, message: String)
    fun getSource(): String
    fun getItems(): List<AddToListItem>
    fun hasNoItems(): Boolean
}